File: nodejs/src/model/CheckTransaction.js
import ApiClient from '../ApiClient';
import CheckAccountType from './CheckAccountType';
import CheckTransactionType from './CheckTransactionType';

export default class CheckTransaction {
    constructor() {
        this.Terminal_ID = '';
        this.Account_Number = '';
        this.Account_Type = new CheckAccountType();
        this.Check_Number = '';
        this.Routing_Number = '';
        this.MICR_Data = '';
        this.First_Name = '';
        this.Last_Name = '';
        this.Company_Name = '';
        this.Address1 = '';
        this.Address2 = '';
        this.City = '';
        this.State = '';
        this.Zip = '';
        this.Phone_Number = '';
        this.DL_Number = '';
        this.DL_State = '';
        this.Courtesy_Card_ID = '';
        this.SSN4 = '';
        this.DOB_Year = '';
        this.Check_Amount = '';
        this.Tran_Type = '';
        this.Proc_ID = '';
        this.IP_Address = '';
        this.CustomerNumber = '';
        this.PONumber = '';
        this.InvoiceNumber = '';
        this.TranNote = '';
        this.EmailAddress = '';
        this.SEC_Code = '';
        this.Proc_Resp = '';
        this.Proc_Code = '';
        this.Proc_Mess = '';
    }

    static constructFromObject(data, obj) {
        if (data) {
            obj = obj || new CheckTransaction();

            if (Object.prototype.hasOwnProperty.call(data, 'Terminal_ID')) {
                obj['Terminal_ID'] = ApiClient.convertToType(data['Terminal_ID'], 'String');
            }
            if (Object.prototype.hasOwnProperty.call(data, 'Account_Number')) {
                obj['Account_Number'] = ApiClient.convertToType(data['Account_Number'], 'String');
            }
            if (Object.prototype.hasOwnProperty.call(data, 'Account_Type')) {
                obj['Account_Type'] = CheckAccountType.constructFromObject(data['Account_Type']);
            }
            if (Object.prototype.hasOwnProperty.call(data, 'Check_Number')) {
                obj['Check_Number'] = ApiClient.convertToType(data['Check_Number'], 'String');
            }
            if (Object.prototype.hasOwnProperty.call(data, 'Routing_Number')) {
                obj['Routing_Number'] = ApiClient.convertToType(data['Routing_Number'], 'String');
            }
            if (Object.prototype.hasOwnProperty.call(data, 'MICR_Data')) {
                obj['MICR_Data'] = ApiClient.convertToType(data['MICR_Data'], 'String');
            }
            if (Object.prototype.hasOwnProperty.call(data, 'First_Name')) {
                obj['First_Name'] = ApiClient.convertToType(data['First_Name'], 'String');
            }
            if (Object.prototype.hasOwnProperty.call(data, 'Last_Name')) {
                obj['Last_Name'] = ApiClient.convertToType(data['Last_Name'], 'String');
            }
            if (Object.prototype.hasOwnProperty.call(data, 'Company_Name')) {
                obj['Company_Name'] = ApiClient.convertToType(data['Company_Name'], 'String');
            }
            if (Object.prototype.hasOwnProperty.call(data, 'Address1')) {
                obj['Address1'] = ApiClient.convertToType(data['Address1'], 'String');
            }
            if (Object.prototype.hasOwnProperty.call(data, 'Address2')) {
                obj['Address2'] = ApiClient.convertToType(data['Address2'], 'String');
            }
            if (Object.prototype.hasOwnProperty.call(data, 'City')) {
                obj['City'] = ApiClient.convertToType(data['City'], 'String');
            }
            if (Object.prototype.hasOwnProperty.call(data, 'State')) {
                obj['State'] = ApiClient.convertToType(data['State'], 'String');
            }
            if (Object.prototype.hasOwnProperty.call(data, 'Zip')) {
                obj['Zip'] = ApiClient.convertToType(data['Zip'], 'String');
            }
            if (Object.prototype.hasOwnProperty.call(data, 'Phone_Number')) {
                obj['Phone_Number'] = ApiClient.convertToType(data['Phone_Number'], 'String');
            }
            if (Object.prototype.hasOwnProperty.call(data, 'DL_Number')) {
                obj['DL_Number'] = ApiClient.convertToType(data['DL_Number'], 'String');
            }
            if (Object.prototype.hasOwnProperty.call(data, 'DL_State')) {
                obj['DL_State'] = ApiClient.convertToType(data['DL_State'], 'String');
            }
            if (Object.prototype.hasOwnProperty.call(data, 'Courtesy_Card_ID')) {
                obj['Courtesy_Card_ID'] = ApiClient.convertToType(data['Courtesy_Card_ID'], 'String');
            }
            if (Object.prototype.hasOwnProperty.call(data, 'SSN4')) {
                obj['SSN4'] = ApiClient.convertToType(data['SSN4'], 'String');
            }
            if (Object.prototype.hasOwnProperty.call(data, 'DOB_Year')) {
                obj['DOB_Year'] = ApiClient.convertToType(data['DOB_Year'], 'String');
            }
            if (Object.prototype.hasOwnProperty.call(data, 'Check_Amount')) {
                obj['Check_Amount'] = ApiClient.convertToType(data['Check_Amount'], 'String');
            }
            if (Object.prototype.hasOwnProperty.call(data, 'Tran_Type')) {
                obj['Tran_Type'] = CheckTransactionType.constructFromObject(data['Tran_Type']);
            }
            if (Object.prototype.hasOwnProperty.call(data, 'Proc_ID')) {
                obj['Proc_ID'] = ApiClient.convertToType(data['Proc_ID'], 'String');
            }
            if (Object.prototype.hasOwnProperty.call(data, 'IP_Address')) {
                obj['IP_Address'] = ApiClient.convertToType(data['IP_Address'], 'String');
            }
            if (Object.prototype.hasOwnProperty.call(data, 'CustomerNumber')) {
                obj['CustomerNumber'] = ApiClient.convertToType(data['CustomerNumber'], 'String');
            }
            if (Object.prototype.hasOwnProperty.call(data, 'PONumber')) {
                obj['PONumber'] = ApiClient.convertToType(data['PONumber'], 'String');
            }
            if (Object.prototype.hasOwnProperty.call(data, 'InvoiceNumber')) {
                obj['InvoiceNumber'] = ApiClient.convertToType(data['InvoiceNumber'], 'String');
            }
            if (Object.prototype.hasOwnProperty.call(data, 'TranNote')) {
                obj['TranNote'] = ApiClient.convertToType(data['TranNote'], 'String');
            }
            if (Object.prototype.hasOwnProperty.call(data, 'EmailAddress')) {
                obj['EmailAddress'] = ApiClient.convertToType(data['EmailAddress'], 'String');
            }
            if (Object.prototype.hasOwnProperty.call(data, 'SEC_Code')) {
                obj['SEC_Code'] = ApiClient.convertToType(data['SEC_Code'], 'String');
            }
            if (Object.prototype.hasOwnProperty.call(data, 'Proc_Resp')) {
                obj['Proc_Resp'] = ApiClient.convertToType(data['Proc_Resp'], 'String');
            }
            if (Object.prototype.hasOwnProperty.call(data, 'Proc_Code')) {
                obj['Proc_Code'] = ApiClient.convertToType(data['Proc_Code'], 'String');
            }
            if (Object.prototype.hasOwnProperty.call(data, 'Proc_Mess')) {
                obj['Proc_Mess'] = ApiClient.convertToType(data['Proc_Mess'], 'String');
            }
        }
        return obj;
    }
}

File: CodeSamples/csharp/src/IO.Swagger/Model/CheckTransaction.cs
using System;
using System.Linq;
using System.IO;
using System.Text;
using System.Text.RegularExpressions;
using System.Collections;
using System.Collections.Generic;
using System.Collections.ObjectModel;
using System.Runtime.Serialization;
using Newtonsoft.Json;
using Newtonsoft.Json.Converters;
using System.ComponentModel.DataAnnotations;
using SwaggerDateConverter = IO.Swagger.Client.SwaggerDateConverter;

namespace IO.Swagger.Model
{
    [DataContract]
    public partial class CheckTransaction :  IEquatable<CheckTransaction>, IValidatableObject
    {
        public CheckTransaction(string terminalID = default(string), string accountNumber = default(string), CheckAccountType accountType = default(CheckAccountType), string checkNumber = default(string), string routingNumber = default(string), string mICRData = default(string), string firstName = default(string), string lastName = default(string), string companyName = default(string), string address1 = default(string), string address2 = default(string), string city = default(string), string state = default(string), string zip = default(string), string phoneNumber = default(string), string dLNumber = default(string), string dLState = default(string), string courtesyCardID = default(string), string sSN4 = default(string), string dOBYear = default(string), string checkAmount = default(string), CheckTransactionType tranType = default(CheckTransactionType), string procID = default(string), string iPAddress = default(string), string customerNumber = default(string), string pONumber = default(string), string invoiceNumber = default(string), string tranNote = default(string), string emailAddress = default(string), string sECCode = default(string), string procResp = default(string), string procCode = default(string), string procMess = default(string))
        {
            this.TerminalID = terminalID;
            this.AccountNumber = accountNumber;
            this.AccountType = accountType;
            this.CheckNumber = checkNumber;
            this.RoutingNumber = routingNumber;
            this.MICRData = mICRData;
            this.FirstName = firstName;
            this.LastName = lastName;
            this.CompanyName = companyName;
            this.Address1 = address1;
            this.Address2 = address2;
            this.City = city;
            this.State = state;
            this.Zip = zip;
            this.PhoneNumber = phoneNumber;
            this.DLNumber = dLNumber;
            this.DLState = dLState;
            this.CourtesyCardID = courtesyCardID;
            this.SSN4 = sSN4;
            this.DOBYear = dOBYear;
            this.CheckAmount = checkAmount;
            this.TranType = tranType;
            this.ProcID = procID;
            this.IPAddress = iPAddress;
            this.CustomerNumber = customerNumber;
            this.PONumber = pONumber;
            this.InvoiceNumber = invoiceNumber;
            this.TranNote = tranNote;
            this.EmailAddress = emailAddress;
            this.SECCode = sECCode;
            this.ProcResp = procResp;
            this.ProcCode = procCode;
            this.ProcMess = procMess;
        }

        [DataMember(Name="Terminal_ID", EmitDefaultValue=false)]
        public string TerminalID { get; set; }

        [DataMember(Name="Account_Number", EmitDefaultValue=false)]
        public string AccountNumber { get; set; }

        [DataMember(Name="Account_Type", EmitDefaultValue=false)]
        public CheckAccountType AccountType { get; set; }

        [DataMember(Name="Check_Number", EmitDefaultValue=false)]
        public string CheckNumber { get; set; }

        [DataMember(Name="Routing_Number", EmitDefaultValue=false)]
        public string RoutingNumber { get; set; }

        [DataMember(Name="MICR_Data", EmitDefaultValue=false)]
        public string MICRData { get; set; }

        [DataMember(Name="First_Name", EmitDefaultValue=false)]
        public string FirstName { get; set; }

        [DataMember(Name="Last_Name", EmitDefaultValue=false)]
        public string LastName { get; set; }

        [DataMember(Name="Company_Name", EmitDefaultValue=false)]
        public string CompanyName { get; set; }

        [DataMember(Name="Address1", EmitDefaultValue=false)]
        public string Address1 { get; set; }

        [DataMember(Name="Address2", EmitDefaultValue=false)]
        public string Address2 { get; set; }

        [DataMember(Name="City", EmitDefaultValue=false)]
        public string City { get; set; }

        [DataMember(Name="State", EmitDefaultValue=false)]
        public string State { get; set; }

        [DataMember(Name="Zip", EmitDefaultValue=false)]
        public string Zip { get; set; }

        [DataMember(Name="Phone_Number", EmitDefaultValue=false)]
        public string PhoneNumber { get; set; }

        [DataMember(Name="DL_Number", EmitDefaultValue=false)]
        public string DLNumber { get; set; }

        [DataMember(Name="DL_State", EmitDefaultValue=false)]
        public string DLState { get; set; }

        [DataMember(Name="Courtesy_Card_ID", EmitDefaultValue=false)]
        public string CourtesyCardID { get; set; }

        [DataMember(Name="SSN4", EmitDefaultValue=false)]
        public string SSN4 { get; set; }

        [DataMember(Name="DOB_Year", EmitDefaultValue=false)]
        public string DOBYear { get; set; }

        [DataMember(Name="Check_Amount", EmitDefaultValue=false)]
        public string CheckAmount { get; set; }

        [DataMember(Name="Tran_Type", EmitDefaultValue=false)]
        public CheckTransactionType TranType { get; set; }

        [DataMember(Name="Proc_ID", EmitDefaultValue=false)]
        public string ProcID { get; set; }

        [DataMember(Name="IP_Address", EmitDefaultValue=false)]
        public string IPAddress { get; set; }

        [DataMember(Name="CustomerNumber", EmitDefaultValue=false)]
        public string CustomerNumber { get; set; }

        [DataMember(Name="PONumber", EmitDefaultValue=false)]
        public string PONumber { get; set; }

        [DataMember(Name="InvoiceNumber", EmitDefaultValue=false)]
        public string InvoiceNumber { get; set; }

        [DataMember(Name="TranNote", EmitDefaultValue=false)]
        public string TranNote { get; set; }

        [DataMember(Name="EmailAddress", EmitDefaultValue=false)]
        public string EmailAddress { get; set; }

        [DataMember(Name="SEC_Code", EmitDefaultValue=false)]
        public string SECCode { get; set; }

        [DataMember(Name="Proc_Resp", EmitDefaultValue=false)]
        public string ProcResp { get; set; }

        [DataMember(Name="Proc_Code", EmitDefaultValue=false)]
        public string ProcCode { get; set; }

        [DataMember(Name="Proc_Mess", EmitDefaultValue=false)]
        public string ProcMess { get; set; }
    }
}
